src/models/CommonModel.ts

```diff
@@ -20,6 +20,102 @@ export class CommonModel extends CommonSchema<CommonModel> {
   extend?: string[];
   originalSchema?: Schema | boolean;
   
+  /**
+   * Retrieves data from originalSchema by given key
+   * 
+   * @param key given key
+   * @returns {any}
+   */
+  getFromSchema<K extends keyof Schema>(key: K) {
+    let schema = this.originalSchema || {};
+    if (typeof schema === 'boolean') schema = {};
+    return schema[`${key}`];
+  }
+
+  /**
+   * Set the types of the model
+   * 
+   * @param types to set the model type to
+   */
+  setType(types : string | string[]) {
+    if (Array.isArray(types)) {
+      if (types.length === 0) {
+        this.type = undefined;
+        return;
+      } else if (types.length === 1) {
+        this.type = types[0];
+        return;
+      }
+    }
+    this.type = types;
+  }
+
+  /**
+   * Adds types to the existing model types.
+   * 
+   * Makes sure to only keep a single type incase of duplicates.
+   * 
+   * @param types which types we should try and add to the existing output
+   */
+  addToTypes(types: string[] | string) {
+    if (Array.isArray(types)) {
+      types.forEach((value) => {
+        this.addToTypes(value);
+      });
+    } else if (this.type === undefined) {
+      this.type = types;
+    } else if (!Array.isArray(this.type) && this.type !== types) {
+      this.type = [this.type, types];
+    } else if (Array.isArray(this.type) && !this.type.includes(types)) {
+      this.type.push(types);
+    }
+  }
+
+  /**
+   * Checks if given property name is required in object
```

Comment:
imho, you duplicate documentation here, you have typescript types and jsdoc types. If you actually need this for something else I am fine with it.

Reply:
We use it to provide a generated API file for the library - https://github.com/asyncapi/generator-model-sdk/blob/master/API.md and it does not imply the type from TS, unfortunately.

Reply:
I have seen generators for something like that but I never actually had to use them. Nevertheless you may find it usefull.

Reply:
For now, let's keep it, but gonna keep it in mind.